docs/distributed_data_loading.md

```diff
@@ -0,0 +1,460 @@
+
+Every {class}`jax.Array` has an associated {mod}`jax.sharding.Sharding`, which describes which shard of the global data is required by each global device. When you create a {class}`jax.Array` from scratch, you also need to create its `Sharding`. This is how JAX can understand how the data is laid out across devices. You can create whatever `Sharding` you want. In practice, you usually pick a `Sharding` based on what kind of parallelism strategy you are implementing (you will learn more about data and model parallelism in more detail later in this guide). You can also pick a `Sharding` based on how the raw data will be produced within each process.
+
+Once you have defined a {mod}`jax.sharding.Sharding`, you can use {func}`~jax.sharding.Sharding.addressable_devices()` to provide a list of devices needed to load data for within the current process. (Note: The term "addressable devices" is a more general version of "local devices". The goal is to make sure that each process's data loader provides the right data to all of that process' local devices.
+
+### Examples
+
+For example, consider a `(64, 128)` {class}`jax.Array` that you need to shard across 4 processes with 2 devices each (8 devices total). This will result in 8 unique data shards, one for each device. There are many ways to shard this {class}`jax.Array`. You can perform a 1D sharding across the second dimension of the {class}`jax.Array`, giving each device a `(64, 16)` shard, as demonstrated below:
+
+<center>
+
+![8 unique data shards](./_static/distributed_data_loading/1.svg)
+
+/_static/distributed_data_loading/
+
+</center>
+
+In the above figure, each data shard has its own color to indicate which process needs to load that shard. For example, you assume process `0`'s 2 devices contain shards `A` and `B`, corresponding to the first `(64, 32)` piece of the global data.
+
+You can pick a different distribution of shards to devices. For example:
+
+<center>
+
+![8 unique data shards - different distribution](./_static/distributed_data_loading/2.svg)
+
+</center>
+
+Here is another example — a 2D sharding:
+
+<center>
+
+![2D sharding](./_static/distributed_data_loading/3.svg)
+
+</center>
+
+However, the {class}`jax.Array` happens to be sharded, you have to make sure that each process's data loader is provided with/loads the required shard(s) of the global data. There are several high-level methods for achieving this: 1) load the Load the global data in each process; 2) use a per-device data pipeline; 3) use a consolidated per-process data pipeline; 4) load data in some convenient way, reshard inside computation
+
+### Option 0: Load the global data in each process
```

Comment:
It's strange that the headings use zero-based indexing, but the parallel list in the paragraph above uses one-based indexing. We should make them match

Reply:
Done.